server/src/main/java/org/elasticsearch/search/lookup/LeafFieldsLookup.java

```diff
@@ -137,21 +135,12 @@ private FieldLookup loadFieldData(String name) {
             cachedFieldData.put(name, data);
         }
         if (data.fields() == null) {
-            List<Object> values;
-            if (TypeFieldType.NAME.equals(data.fieldType().name())) {
-                values = new ArrayList<>(1);
-                final DocumentMapper mapper = mapperService.documentMapper();
-                if (mapper != null) {
-                    values.add(mapper.type());
-                }
```

Comment:
Is this what you meant @romseygeek ? Are we sure that we can drop this? Looks like a breaking change to me, wondering if it's worth leaving it around although it is constant, not sure.

Reply:
It is a breaking change, but I think it comes under the general umbrella of types removal so probably doesn't need a specific call-out.  If you try and use `_type` in a script in 7x you get a deprecation warning so I think it's fine to remove it.

Reply:
the problem here may be that the field type does come up by name, but then it fails to retrieve values from lucene, or is that ok?

Reply:
I see what you mean.  Maybe we should be removing the special-casing in MapperService.fieldType() in master as well, given that it's deprecated generally in 7.x. @jtibshirani what do you think?

Reply:
I just caught up on the code and was surprised to see we still have `TypeFieldType` in master. Is there a reason we added [this special logic](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/mapper/MapperService.java#L406-L408) instead of removing `TypeFieldType` entirely?

Reply:
I think I added this as a stop-gap while backporting the various type field deprecations.  I'll open a PR to remove it entirely in master.

Reply:
One question I had -- does removing this have an implication for our API version compatibility efforts? I'm not up-to-date on our collaborations there.

Reply:
Version compatibility only applies to the 'shape' of the request, as I understand it.  So we need to support `type` queries, but we don't need to support queries against the field `_type` (otherwise we'd never be able to deprecate types at all...)

Reply:
thanks for the input. It looks like this can go in then, and as a follow-up we will also remove TypeFieldType completely, which sounds like a good plan to me.